backend/tracim_backend/lib/webdav/dav_provider.py

```diff
@@ -89,84 +203,34 @@ def current_workspace(self):
         if you are editing content 21 in workspace 3,
         current_workspace will be 3.
         """
-        return self._generate_if_none(
-            self._current_workspace, self._get_workspace, self._get_current_workspace_label,
-        )
-
-    def _get_workspace(self, workspace_id_fetcher):
-        workspace_id = workspace_id_fetcher()
-        wapi = WorkspaceApi(
-            current_user=self.current_user,
-            session=self.dbsession,
-            config=self.app_config,
-            show_deleted=True,
-        )
-        return wapi.get_one_by_label(workspace_id)
-
-    def _get_current_workspace_label(self) -> str:
-        return webdav_convert_file_name_to_bdd(self.path.split("/")[1])
+        return self.processed_path.current_workspace
 
     @property
     def current_content(self):
```

Comment:
Can you put the typing here?
(it is typing.Optional[X], with some X, if I understand the code correctly)

Reply:
done.